File: oc/CALayerDemo/CALayerDemo/CustomLayer.m
//
//  CustomLayer.m
//  CALayerDemo
//
//  Created by dfang on 2018-2-23.
//  Copyright © 2018年 east. All rights reserved.
//

#import "CustomLayer.h"

@implementation CustomLayer

// UIView在显示的时候，会先调用drawRect方法，然后调用图层的drawInContext
- (void)drawInContext:(CGContextRef)ctx {
    CGContextFillEllipseInRect(ctx, CGRectMake(0, 0, 100, 100));
}

@end

File: swift/CALayerDemo/CALayerDemo/CustomLayer.swift
//
//  CustomLayer.swift
//  CALayerDemo
//
//  Created by dfang on 2018-3-5.
//  Copyright © 2018年 east. All rights reserved.
//

import UIKit

class CustomLayer: CALayer {
    
    override func draw(in ctx: CGContext) {
        ctx.fillEllipse(in: CGRect.init(x: 0, y: 0, width: 100, height: 100))
    }
}
